integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/ingest/CreateElasticsearchBackedHiveTable.java

```diff
@@ -0,0 +1,285 @@
+package com.thinkbiganalytics.nifi.v2.ingest;
+
+/*-
+ * #%L
+ * thinkbig-nifi-elasticsearch-processors
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.thinkbiganalytics.nifi.v2.thrift.ExecuteHQLStatement;
+import com.thinkbiganalytics.nifi.v2.thrift.ThriftService;
+import com.thinkbiganalytics.util.ColumnSpec;
+import com.thinkbiganalytics.util.TableType;
+
+import static com.thinkbiganalytics.nifi.v2.ingest.IngestProperties.FIELD_SPECIFICATION;
+import static com.thinkbiganalytics.nifi.v2.ingest.IngestProperties.FEED_CATEGORY;
+import static com.thinkbiganalytics.nifi.v2.ingest.IngestProperties.FEED_NAME;
+
+
+/**
+ * This processor indexes json data in elasticsearch
+ */
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@Tags({"elasticsearch", "thinkbig", "hive"})
+@CapabilityDescription("Creates a table in Hive that is backed by data in Elasticsearch")
+public class CreateElasticsearchBackedHiveTable extends ExecuteHQLStatement {
+
+
+    public static final Relationship REL_CREATE_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("Successfully created Elasticsearch backed Hive table.")
+        .build();
+
+
+    public static final Relationship REL_CREATE_FAILURE = new Relationship.Builder()
```

Comment:
Would it make sense to remove the relationships defined in in both processors and always just reference the generic ingestProperties.REL_FAILURE? This would, however, provide a less specific description.